x-pack/plugins/apm/server/routes/settings/custom_link.ts

```diff
@@ -4,13 +4,27 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 import * as t from 'io-ts';
-import { FilterOptionsRt } from '../../../common/custom_link_filter_options';
+import { pick } from 'lodash';
+import {
+  SERVICE_NAME,
+  SERVICE_ENVIRONMENT,
+  TRANSACTION_NAME,
+  TRANSACTION_TYPE
+} from '../../../common/elasticsearch_fieldnames';
 import { createRoute } from '../create_route';
 import { setupRequest } from '../../lib/helpers/setup_request';
 import { createOrUpdateCustomLink } from '../../lib/settings/custom_link/create_or_update_custom_link';
 import { deleteCustomLink } from '../../lib/settings/custom_link/delete_custom_link';
 import { listCustomLinks } from '../../lib/settings/custom_link/list_custom_links';
 import { getTransaction } from '../../lib/settings/custom_link/get_transaction';
+import { FILTER_OPTIONS } from '../../../common/custom_link_filter_options';
+
+export const FilterOptionsRt = t.partial({
+  [SERVICE_NAME]: t.string,
+  [SERVICE_ENVIRONMENT]: t.string,
+  [TRANSACTION_NAME]: t.string,
+  [TRANSACTION_TYPE]: t.string
+});
```

Comment:
What about keeping this in custom_link_types.d.ts with the other types?

Reply:
I don't think is a good idea since I moved the `custom_link_types.d.ts` to `common` directory

Reply:
But I think it makes sense to have `custom_link_types.d.ts` in the server as well, to store types that are not shared.